### PR TITLE
Fix irrelevant error messages about log upload during shutdown for gitops run

### LINKED ISSUE #3064: gitops run: Irrelevant error messages about log upload during shutdown
Sometimes, after hitting ctrl+c to turn off gitops run, after every line in the terminal output there's another line printing
```
✗ failed to put log to s3: Put "http://localhost:45423/gitops-run-logs/no-session/20221125-124034.74331.txt": dial tcp [::1]:45423: connect: connection refused
```

I expect the log sender to stop sending logs at the same time as the S3 bucket is shut down.

### PR DESCRIPTION
**Pull request by chanwit**
Fixes #3064 

Signed-off-by: Chanwit Kaewkasi <chanwit@weave.works>

### CODE REVIEW

File: cmd/gitops/beta/run/cmd.go [hunk 1]
```diff
@@ -548,6 +548,18 @@ func runCommandWithoutSession(cmd *cobra.Command, args []string) error {
 	// 2. log is the S3 logger that also delegates its outputs to "log0".
 	log0 := logger.NewCLILogger(os.Stdout)
 
+	state, err := term.GetState(int(os.Stdin.Fd()))
```
Comment (makkes) [inline cmd/gitops/beta/run/cmd.go:551]: Do you know exactly why this is necessary? I've never ran across terminal issues after exiting gitops run and I'd like us to be aligned on why the terminal state isn't as expected when exiting.

Comment (chanwit) [reply cmd/gitops/beta/run/cmd.go:551]: I pressed Ctrl +C many times and was hit by the bug. It's not so hard to reproduce (but still a random behaviour).

Also, Paul Curtis reported it here https://github.com/weaveworks/weave-gitops/issues/3203

Comment (makkes) [reply cmd/gitops/beta/run/cmd.go:551]: Yes, I've seen it being reported but I'd like to understand the circumstances under which it occurs.

**Comment by makkes** [comment]
as discussed in standup, let's please split the PR up into two so that we get proper release notes entries.

**Review by makkes — APPROVED** [review]

**Pull request merged**